packages/client/hmi-client/src/workflow/ops/model-from-code/tera-model-from-code.vue

```diff
@@ -67,14 +67,32 @@
 			<!--Notebook section if we decide we need one-->
 		</div>
 		<template #preview>
-			<tera-drilldown-preview v-model:output="selectedModelFramework" :options="modelFrameworks">
-				<tera-progress-spinner v-if="isProcessing" :font-size="2" />
+			<tera-drilldown-preview>
+				<tera-progress-spinner
+					v-if="isProcessing"
+					:font-size="2"
+					is-centered
+					style="height: 100%"
+				/>
+				<tera-operator-placeholder-graphic
+					v-if="!isProcessing && !selectedModel && !previewHTML"
+					:operation-type="node.operationType"
+					style="height: 100%"
+				/>
 				<section>
-					<template v-if="selectedModelFramework === ModelFramework.Petrinet && selectedModel">
+					<template
+						v-if="
```

Comment:
You could move the `v-if="selectedModel && !isProcessing"` to the `<section>`
Maybe something like the following. What do you think?
```vue
<section v-if="selectedModel && (!isProcessing || !previewHTML)"?>
```

Reply:
I'll probably clean up this logic a bit once we know what we should be showing between the Petrinet & Decapodes views...I think the logic here wouldn't show the Decapodes view since that depends on the previewHTML.  I will definitely be keeping an eye on this in later PRs